go/vm/geth/geth.go

```diff
@@ -101,9 +101,18 @@ func createGethInterpreterContext(parameters vm.Parameters) (*geth.EVM, *geth.Co
 	// Set hard forks for chainconfig
 	chainConfig := *params.AllEthashProtocolChanges
 	chainConfig.ChainID = new(big.Int).SetBytes(context.ChainID[:])
-	chainConfig.IstanbulBlock = big.NewInt(int64(vm.R07_Istanbul) * 10)
-	chainConfig.BerlinBlock = big.NewInt(int64(vm.R09_Berlin) * 10)
-	chainConfig.LondonBlock = big.NewInt(int64(vm.R10_London) * 10)
+	chainConfig.IstanbulBlock = big.NewInt(int64(vm.R07_Istanbul) * 1000)
+	chainConfig.BerlinBlock = big.NewInt(int64(vm.R09_Berlin) * 1000)
+	chainConfig.LondonBlock = big.NewInt(int64(vm.R10_London) * 1000)
+	if parameters.Revision >= vm.R11_Paris {
+		chainConfig.MergeNetsplitBlock = big.NewInt(int64(vm.R11_Paris) * 1000)
+	}
+	if parameters.Revision >= vm.R12_Shanghai {
+		chainConfig.ShanghaiTime = verkleTime(vm.R12_Shanghai)
```

Comment:
Like for the fork block, we should define a `GetForkTime(revision Revision)` in the CT common directory and then use this function to determine the time to be set here.